File: Sesion 7/Marco teorico/3miembros_protected/3.java
import java.util.*;

class Persona {
    protected String nombre;
    protected int edad;

    public Persona(String _nombre, int _edad) {
        this.nombre = _nombre;
        this.edad = _edad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " y tengo " + edad + " años.");
    }
}

class Estudiante extends Persona {
    private int numero_estudiante;

    public Estudiante(String _nombre, int _edad, int _numero_estudiante) {
        super(_nombre, _edad);
        this.numero_estudiante = _numero_estudiante;
    }

    public void mostrarInfo() {
        System.out.println("Soy el estudiante número " + numero_estudiante + ".");
        System.out.println("Mi nombre es: " + nombre + " y tengo " + edad + " años.");
    }
}

public class Main {
    public static void main(String[] args) {
        Estudiante estudiante = new Estudiante("Juan", 20, 12345);
        estudiante.saludar(); 
        estudiante.mostrarInfo(); 
    }
}

File: Sesion 7/Marco teorico/3miembros_protected/3.py
class Persona:
    def __init__(self, _nombre, _edad):
        self.nombre = _nombre
        self.edad = _edad

    def saludar(self):
        print("Hola, soy", self.nombre, "y tengo", self.edad, "años.")

class Estudiante(Persona):
    def __init__(self, _nombre, _edad, _numero_estudiante):
        super().__init__(_nombre, _edad)
        self.numero_estudiante = _numero_estudiante

    def mostrarInfo(self):
        print("Soy el estudiante número", self.numero_estudiante, ".")
        print("Mi nombre es:", self.nombre, "y tengo", self.edad, "años.")


estudiante = Estudiante("Juan", 20, 12345)
estudiante.saludar() 
estudiante.mostrarInfo()  